src/components/Card.js

```diff
@@ -1,35 +1,60 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleCardClick) {
-    this._name = name;
-    this._link = link;
+  constructor(
+    data,
+    cardSelector,
+    userId,
+    handleCardClick,
+    handleCardLike,
+    handleCardDelete
+  ) {
+    this._name = data.name;
+    this._link = data.link;
+    this._id = data._id;
+    this._ownerId = data.owner._id;
+    this.likes = data.likes;
+    this._userId = userId;
+    this.isLiked = this.checkIfLiked();
+
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this.handleCardLike = handleCardLike;
+    this._handleCardDelete = handleCardDelete;
+  }
+
+  checkIfLiked() {
+    const liked = this.likes.some((card) => card._id === this._userId);
+    return liked;
+  }
+
+  _updateLikeIcon() {
+    this._likeButton.classList.toggle("card__like-button_active");
+    return (this.isLiked = !this.isLiked);
   }
 
   _setEventListeners() {
     this._likeButton = this._cardElement.querySelector(".card__like-button");
+    if (this.isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    }
     this._deleteButton = this._cardElement.querySelector(
       ".card__delete-button"
     );
 
-    this._likeButton.addEventListener("click", () => this._handleLikeIcon());
+    if (this._userId !== this._ownerId) {
+      this._deleteButton.remove();
+    }
+    this._likeButton.addEventListener("click", () => {
+      this.handleCardLike(this._id, this.isLiked, this.cardLikeCount);
+      this._updateLikeIcon();
```

Comment:
`this._updateLikeIcon();`


You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.